core.ts

```diff
@@ -644,17 +644,14 @@ function PinejsClientCoreFactory(utils: PinejsClientCoreFactory.Util, Promise: P
 				let queryOptions: string[] = []
 				if (params.options != null) {
 					queryOptions = mapObj(params.options, (value, option) => {
-						if (option[0] !== '$') {
-							deprecated.implicitOptions()
-							option = '$' + option
-						}
-						if(!isValidOption(option)) {
-							throw new Error(`Unknown option '${option}'`)
+						if(option[0] === '$' && !isValidOption(option)) {
+							throw new Error(`Unknown odata option '${option}'`)
```

Comment:
Previously this would throw an error if you provided an invalid option, and now it doesn't if they don't start with `$`. Is that correct? Why not just throw for all invalid options, including unprefixed ones?

Reply:
Previously we automatically prefixed the $, so we already knew every option started with a $, and $ prefixes are reserved for odata options and so can be checked for validity.  Without a prefix then any option is allowed, which is why `customOptions` doesn't have the check